### PR TITLE
Add custom segmentation properties

### PR DESCRIPTION
**Pull request by xitzhang**
**1. Add Speech_SegmentationStrategy:**
The strategy used to determine when a spoken phrase has ended and a final Recognized result should be generated.
Allowed values:
**Default**: Delegates the strategy selection to the Speech Service. Mapped to "Normal" segmentation mode.
**Time**: Represents the current behavior. Mapped to "Custom" segmentation mode. Users can adjust segmentation by using “Speech_SegmentationSilenceTimeoutMs” and “Speech_SegmentationMaximumTimeMs”.
**Semantic**: Mapped to "Semantic" segmentation mode.

**2. Add Speech_SegmentationMaximumTimeMs**
When using the time strategy, the <see cref="Speech_SegmentationSilenceTimeoutMs"/> property can be used to adjust the amount of silence needed to determine the end of a spoken phrase
This property was not disabled in "Semantic" segmentation strategy.

To prevent buffer overflow when users configure Speech_SegmentationMaximumTimeMs, adjust the audio buffer size to 2.2 times the value of Speech_SegmentationMaximumTimeMs if set.

**Tests:**
**AutoSourceLangDetectionTests**:
Test Suites: 2 passed, 2 total
Tests:       8 skipped, 20 passed, 28 total
Snapshots:   0 total
Time:        58.593 s, estimated 67 s
Ran all test suites in 2 projects.
**SpeechConfigTests.ts**:
Test Suites: 2 passed, 2 total
Tests:       5 skipped, 167 passed, 172 total
Snapshots:   0 total
Time:        91.874 s
Ran all test suites in 2 projects.
**ConversationTranscriberTests.ts**:
Test Suites: 2 passed, 2 total
Tests:       26 passed, 26 total
Snapshots:   0 total
Time:        147.339 s
Ran all test suites in 2 projects.

### CODE REVIEW

File: tests/SpeechConfigTests.ts [hunk 1]
```diff
@@ -877,6 +877,40 @@ describe("Connection URL Tests", (): void => {
...
+                        (s: sdk.SpeechConfig): void => {
+                            s.setProperty(sdk.PropertyId[sdk.PropertyId.Speech_SegmentationMaximumTimeMs], val);
+                        },
+                        recognizerCreateMethod,
+                        done,
+                        propName + "=" + val
+                    );
+                });
+                
+                test("segmentationStrategy", (done: jest.DoneCallback): void => {
```
Comment (rhurey) [inline tests/SpeechConfigTests.ts:897]: This is winding up in the URL instead of just in the context message?

Comment (xitzhang) [reply tests/SpeechConfigTests.ts:897]: Yes, I found existing slienceTimeOut in query url and thus I also added maxSegmentTimeout and strategy to url

**Review by rhurey — APPROVED** [review]

**Pull request merged**